### PR TITLE
Simplify node selection

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch7591]

File: src/components/NetworkNodeDropdown.tsx [hunk 1]
```diff
@@ -109,18 +109,18 @@ const NetworkNodeDropdown: FC<Props> = ({ networkId, onEdit }) => {
...
-  const { nodes, autoNode: autoNodeName } = network;
+  const { nodes } = network;
   const autoNode = {
-    ...NetworkUtils.getAutoNode(network),
     service: autoNodeLabel
   };
-  const { service, name: selectedNodeName } = selectedNode;
+  const selectedNode = _selectedNode || autoNode;
+  const { service } = selectedNode;
   const displayNodes = [autoNode, ...nodes, ...(isFunction(onEdit) ? [{ service: newNode }] : [])];
```
[the remark this answers is not in the document]
Comment (vilsbole) [reply src/components/NetworkNodeDropdown.tsx:118]: In fact a `Selector` has a `getOptionLabel` prop which removes the need to remap options to { label, value } so we don't need to destructure `selectedNode` and we can write:

```
<Selector<NodeOptions & any>
  getOptionLabel={(n) => n.service || n.name}
  value={selectedNode}
  options={displayNodes}
  ...
```

The change is small, it removes a superfluous `map` + intermediate variables and only requires remapping the displayed properties in `optionComponent` and `valueComponent`. A useful example of this technique is `AssetSelector`.

**Review by vilsbole — CHANGES_REQUESTED** [review]

File: src/services/Store/store/persist.config.ts [hunk 2]
```diff
@@ -130,7 +131,22 @@ const customDeserializer = (slice: ValuesType<LocalStorage>) => {
   }
 };
 
+const migrations = {
+  2: (state: DataStore) => {
```
[the remark this answers is not in the document]
Comment (FrederikBolding) [reply src/services/Store/store/persist.config.ts:135]: The API requires a `number`, so this is impossible.

File: src/services/Store/store/persist.config.ts [hunk 3]
```diff
@@ -130,7 +131,22 @@ const customDeserializer = (slice: ValuesType<LocalStorage>) => {
...
+    return {
+      ...state,
+      networks: state.networks.map((n) => ({
+        ...n,
+        autoNode: undefined,
```
[the remark this answers is not in the document]
Comment (FrederikBolding) [reply src/services/Store/store/persist.config.ts:140]: Sure, is there a pretty way to do it in a map? I couldn't think of one, hence I just used `undefined`.

File: src/components/NetworkNodeDropdown.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (FrederikBolding) [reply src/components/NetworkNodeDropdown.tsx:118]: I don't agree entirely with your simplification of removing `autoNode` fx. I think it is clearer with that. But I have simplified further.

File: src/services/Store/store/persist.config.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (vilsbole) [reply src/services/Store/store/persist.config.ts:135]: Ok 👍

File: src/services/Store/store/persist.config.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (vilsbole) [reply src/services/Store/store/persist.config.ts:140]: In es6 destructuring works well for this.
```
networks: state.networks.map(({ autoNode, ...n }) => ({ ...n })
```

[the remark this answers is not in the document]
Comment (FrederikBolding) [reply src/services/Store/store/persist.config.ts:140]: Duh, thanks 😆

**Review by vilsbole — APPROVED** [review]

**Pull request merged**